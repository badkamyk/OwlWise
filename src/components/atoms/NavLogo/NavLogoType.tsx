export type NavLogoType = {
	href: string;
	alt: string;
	text: string;
	src: string;
};
