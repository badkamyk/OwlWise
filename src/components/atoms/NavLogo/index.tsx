import { NavLogoType } from "./NavLogoType";

export default function NavLogo({ href, src, alt, text }: NavLogoType) {
	return (
		<a href={href} className="flex items-center" data-testid="nav-logo">
			<img src={src} className="mr-3 h-6 sm:h-9" alt={alt} />
			<span className="self-center whitespace-nowrap text-xl font-semibold dark:text-white">
				{text}
			</span>
		</a>
	);
}
