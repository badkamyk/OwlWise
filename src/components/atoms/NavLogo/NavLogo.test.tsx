import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import NavLogo from "./index";

it("renders rNavLogo with props correctly", () => {
	const { container } = render(
		<NavLogo href="/" src="/favicon.ico" alt="Logo" text="Logo" />
	);
	expect(container.getElementsByClassName("flex items-center")).toBeTruthy();
	expect(container.getElementsByClassName("mr-3 h-6 sm:h-9")).toBeTruthy();
	expect(screen.getByRole("link")).toHaveProperty("href", "/");
	expect(
		container.getElementsByClassName(
			"self-center whitespace-nowrap text-xl font-semibold dark:text-white"
		)
	).toBeTruthy();
	expect(screen.getByText("Logo")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
