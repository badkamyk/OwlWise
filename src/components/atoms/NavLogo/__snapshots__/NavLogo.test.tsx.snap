// Vitest Snapshot v1

exports[`NavLogo with props 1`] = `
<div>
  <a
    class="flex items-center"
    href="/"
  >
    <img
      alt="Logo"
      class="mr-3 h-6 sm:h-9"
      src="/favicon.ico"
    />
    <span
      class="self-center whitespace-nowrap text-xl font-semibold dark:text-white"
    >
      Logo
    </span>
  </a>
</div>
`;

exports[`renders rNavLogo with props correctly 1`] = `
<div>
  <a
    class="flex items-center"
    href="/"
  >
    <img
      alt="Logo"
      class="mr-3 h-6 sm:h-9"
      src="/favicon.ico"
    />
    <span
      class="self-center whitespace-nowrap text-xl font-semibold dark:text-white"
    >
      Logo
    </span>
  </a>
</div>
`;
